src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,243 @@
+
+        User actualUser3 = new User();
+        actualUser3.setLogin("A");
+        actualUser3.setPassword("344gs&ws");
+        actualUser3.setAge(39);
+
+        assertThrows(DataCorrectnessException.class,
+                () -> registrationService.register(actualUser3));
+
+        User actualUser4 = new User();
+        actualUser4.setLogin("");
+        actualUser4.setPassword("344gs&ws");
+        actualUser4.setAge(39);
+
+        assertThrows(DataCorrectnessException.class,
+                () -> registrationService.register(actualUser4));
+    }
+
+    @Test
+    void register_passwordAllowableSize_notOk() {
+        User actualUser1 = new User();
+        actualUser1.setLogin("Asallre");
+        actualUser1.setPassword("344gs");
+        actualUser1.setAge(39);
+
+        assertThrows(DataCorrectnessException.class,
+                () -> registrationService.register(actualUser1));
+
+        User actualUser2 = new User();
+        actualUser2.setLogin("Dsterres");
+        actualUser2.setPassword("34");
+        actualUser2.setAge(39);
+
+        assertThrows(DataCorrectnessException.class,
+                () -> registrationService.register(actualUser2));
+
+        User actualUser3 = new User();
+        actualUser3.setLogin("Roparked");
+        actualUser3.setPassword("3");
+        actualUser3.setAge(39);
+
+        assertThrows(DataCorrectnessException.class,
+                () -> registrationService.register(actualUser3));
+
+        User actualUser4 = new User();
+        actualUser4.setLogin("Wlrensar");
+        actualUser4.setPassword("");
+        actualUser4.setAge(39);
+
+        assertThrows(DataCorrectnessException.class,
+                () -> registrationService.register(actualUser4));
+    }
+
+    @Test
+    void register_AllowableAge_notOk() {
+        User actualUser1 = new User();
+        actualUser1.setLogin("Asallre");
+        actualUser1.setPassword("3gd4h8@44gs");
+        actualUser1.setAge(17);
+
+        assertThrows(DataCorrectnessException.class,
+                () -> registrationService.register(actualUser1));
+
+        User actualUser2 = new User();
+        actualUser2.setLogin("Dsterres");
+        actualUser2.setPassword("3gd4h8@44gs");
+        actualUser2.setAge(4);
+
+        assertThrows(DataCorrectnessException.class,
+                () -> registrationService.register(actualUser2));
+
+        User actualUser3 = new User();
+        actualUser3.setLogin("Roparked");
+        actualUser3.setPassword("3gd4h8@44gs");
+        actualUser3.setAge(0);
+
+        assertThrows(DataCorrectnessException.class,
+                () -> registrationService.register(actualUser3));
+
+        User actualUser4 = new User();
+        actualUser4.setLogin("Wlrensar");
+        actualUser4.setPassword("3gd4h8@44gs");
+        actualUser4.setAge(-5);
+
+        assertThrows(DataCorrectnessException.class,
+                () -> registrationService.register(actualUser4));
+
```

Comment:
```suggestion
```

Reply:
Sorry but, I don't quite understand what this trailer means